gobblin-core/src/main/java/gobblin/publisher/HiveRegistrationPublisher.java

```diff
@@ -74,7 +76,15 @@
   private final HiveRegister hiveRegister;
   private final ExecutorService hivePolicyExecutor;
   private final MetricContext metricContext;
-
+  /**
+   * Make the deduplication of path to be registered in the Publisher level,
+   * So that each invocation of {@link #publishData(Collection)} contribute paths registered to this set.
+   */
+  private static Set<String> pathsToRegisterFromSingleState = Sets.newHashSet();
```

Comment:
Is this optimization safe? What happens if new files with an updated schema are published to the same path?